src/sentry/integrations/jira/integration.py

```diff
@@ -232,6 +241,93 @@ def get_create_issue_config(self, group, **kwargs):
 
         return fields
 
+    def create_issue(self, data, **kwargs):
+        client = self.get_client()
+        cleaned_data = {}
+        # protect against mis-configured plugin submitting a form without an
+        # issuetype assigned.
+        if not data.get('issuetype'):
+            raise IntegrationError('Issue Type is required.')
+
+        jira_project = data.get('project')
+        if not jira_project:
+            raise IntegrationError('JIRA project is required.')
+
+        meta = client.get_create_meta_for_project(jira_project)
+
+        if not meta:
+            raise IntegrationError('Something went wrong. Check your plugin configuration.')
+
+        issue_type_meta = self.get_issue_type_meta(data['issuetype'], meta)
+
+        fs = issue_type_meta['fields']
+        for field in fs.keys():
+            f = fs[field]
+            if field == 'description':
+                cleaned_data[field] = data[field]
+                continue
+            elif field == 'summary':
+                cleaned_data['summary'] = data['title']
+                continue
+            if field in data.keys():
+                v = data.get(field)
+                if v:
+                    schema = f.get('schema')
+                    if schema:
```

Comment:
nit: Also could be a guard statment

```py
if not schema:
    cleaned_data[field] = v
    continue
```

Reply:
i agree this is gross, but with your suggestion, we'd have `cleaned_data[field] = v` twice which i'm not sure is that much better? also this is all copied so just don't want to spend a ton of time cleaning stuff up